src/main/java/org/reaktivity/nukleus/oauth/internal/stream/OAuthProxyFactory.java

```diff
@@ -707,6 +821,30 @@ private void onTokenExpiredSignal(
             }
         }
 
+        private void onTokenExpiringSoonSignal(
+            SignalFW signal)
```

Comment:
`signal` is only used for the `traceId` so rename this method to `doChallenge(traceId)` instead